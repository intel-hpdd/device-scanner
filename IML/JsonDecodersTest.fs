module IML.JsonDecodersTest

open IML.JsonDecoders
open Fable.PowerPack
open Fable.Import.Jest
open Matchers

let private ofString = Json.ofString >> Result.unwrapResult
let private plainMap = ofString """{}"""

test "unwrap object" <| fun () ->
  expect.assertions(2)

  toMatchSnapshot (unwrapObject plainMap)

  expect.Invoke(fun () -> unwrapObject (ofString """null""")).toThrowErrorMatchingSnapshot()

test "object" <| fun () ->
  expect.assertions(2)

  toMatchSnapshot (object plainMap)
  toMatchSnapshot (object (ofString """null"""))

test "str" <| fun () ->
  expect.assertions(2)

  toMatchSnapshot (str (Json.String "foo"))
  toMatchSnapshot (str (ofString """null"""))

test "unwrap string" <| fun () ->
  expect.assertions(2)

  toMatchSnapshot (unwrapString (Json.String "bar"))

  expect.Invoke(fun () -> unwrapString (Json.Boolean true)).toThrowErrorMatchingSnapshot()
