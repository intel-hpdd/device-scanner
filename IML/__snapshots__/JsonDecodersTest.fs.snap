// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object 1`] = `
FableMap {
  "comparer": Comparer {
    "Compare": [Function],
  },
  "tree": MapTree {
    "data": undefined,
    "tag": 0,
  },
}
`;

exports[`object 2`] = `null`;

exports[`str 1`] = `"foo"`;

exports[`str 2`] = `null`;

exports[`unwrap object 1`] = `
FableMap {
  "comparer": Comparer {
    "Compare": [Function],
  },
  "tree": MapTree {
    "data": undefined,
    "tag": 0,
  },
}
`;

exports[`unwrap object 2`] = `"Invalid JSON, it must be an object"`;

exports[`unwrap string 1`] = `"bar"`;

exports[`unwrap string 2`] = `"Invalid JSON, it must be a string"`;
